internal/kafka/internal/services/data_plane_kafka.go

```diff
@@ -150,14 +151,30 @@ func (d *dataPlaneKafkaService) processRealKafkaDeployment(ks *dbapi.DataPlaneKa
 		log.Warningf("clusterId for kafka cluster %s does not match clusterId. kafka clusterId = %s :: clusterId = %s", kafka.ID, kafka.ClusterID, cluster.ClusterID)
 		return
 	}
+
+	// Notes on state transitions
+	//  - 'suspending' state is set by the user from a 'ready' state. This must only transition to 'resuming', 'suspended' or 'deprovision'.
+	//     - FSO will only send the status 'suspended' if the Kafka instance was already in a 'suspending' or 'suspended' state.
+	//     - FSO will not change the status of the ManagedKafka CR prior to transitioning to 'suspended' unless an error occurs.
+	//       e.g. If the Kafka instance was in a 'ready' state previously, FSO will keep reporting its status as 'ready' until it
+	//            finally reports 'suspended'. If an error occurs, the 'Ready' condition of the CR will have the values 'Status=False,Reason=Error'.
+	//  - 'resuming' state is set by the user from a 'suspending' or 'suspended' state. This must only transition to 'ready', 'failed' or 'deprovision'.
```

Comment:
technically resuming can't be set by any user (only via admin endpoint). do you think we could rewrite it a little?

Reply:
good point, will update this :)